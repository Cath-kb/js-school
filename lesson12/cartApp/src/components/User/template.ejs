<li class="mdl-list__item demo-user-block__name mdl-color--primary">
    <span class="mdl-list__item-primary-content mdl-typography--headline mdl-color-text--white"><%= user.name %></span>
</li>
<li class="mdl-list__item">
    <span class="mdl-list__item-primary-content"><%= user.email %></span>
</li>
<li class="mdl-list__item">
    <span class="mdl-list__item-primary-content"><%= user.phone %></span>
</li>
<li class="mdl-list__item">
    <span class="mdl-list__item-primary-content"><%= user.address %>, <%= user.city %>, <%= user.country %></span>
</li>
