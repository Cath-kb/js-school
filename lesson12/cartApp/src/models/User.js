import api from './api';

const DEFAULT_USER = 1;

export async function getUser(userId = DEFAULT_USER) {
    try {
        const { data } = await api.get(`/users/${userId}`);

        return data;
    } catch (e) {
        throw e; // todo: handle error
    }
}