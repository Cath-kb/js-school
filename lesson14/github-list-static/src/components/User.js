import React from 'react';

class User extends React.Component {
  render() {
    const {avatar, person} = this.props.user;

    return (
      <div className="row my-4">
        <div className="col-sm-2" data-id="user">
          <img src={avatar.url} alt="User Avatar" className="w-100 img-fluid rounded mb-4 mb-sm-0" />
        </div>
        <div className="col-sm-10">
          <h2 className="d-flex align-items-center">
            {person.login}
            {person.isAdmin && <span className="ml-4 badge badge-pill badge-dark">admin</span>}
          </h2>
          <h3><a href={person.url} target="_blank" title="Go to user profile">{person.url}</a></h3>
        </div>
      </div>
    );
  }
}

export default User;
