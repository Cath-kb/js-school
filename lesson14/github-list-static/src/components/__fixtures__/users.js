const users = [
  {
    id: 1,
    avatar: {
      url: "https://avatars0.githubusercontent.com/u/18?v=4",
    },
    person: {
      login: "wayneeseguin",
      url: "https://api.github.com/users/wayneeseguin",
      isAdmin: false,
    }
  },
  {
    id: 2,
    avatar: {
      url: "https://avatars0.githubusercontent.com/u/19?v=4",
    },
    person: {
      login: "brynary",
      url: "https://api.github.com/users/brynary",
      isAdmin: true,
    }
  }
];

export default users;