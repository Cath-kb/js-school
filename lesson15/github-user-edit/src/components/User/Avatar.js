import React from "react";

const Avatar = ({avatar}) => (
  <img
    className="img-fluid mb-4 rounded"
    src={avatar.url}
    alt="User Avatar"
  />
);

export default Avatar;
