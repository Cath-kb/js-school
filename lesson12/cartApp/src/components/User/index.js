import template from './template.ejs';

export default ({name, email, phone, address, city, country}) => template({
  user: {name, email, phone, address, city, country},
});
