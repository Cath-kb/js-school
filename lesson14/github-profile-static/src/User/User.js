import React from "react";
import Avatar from "./Avatar";
import Person from "./Person";
import Statistics from "./Statistics";

class User extends React.Component {
  render() {
    const {avatar, person, statistics} = this.props.user;
    return (
      <div className="row">
        <div className="col-md-6  mb-4">
          <Avatar avatar={avatar} />
        </div>
        <div className="col-md-6">
          <Person person={person} />
          <Statistics data={statistics} />
        </div>
      </div>
    );
  }
}

export default User;
