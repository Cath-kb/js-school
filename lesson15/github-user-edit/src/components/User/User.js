import React from 'react';

import Avatar from "./Avatar";
import Person from "./Person";
import Statistics from "./Statistics";

class User extends React.Component {
  render() {
    const { onClick } = this.props;
    const {avatar, person, statistics } = this.props.user;

    return (
      <div className="row">
        <div className="col-md-4 text-center text-md-left mb-4">
          <Avatar avatar={avatar} />
        </div>
        <div className="col-md-8">
          <Person person={person} onClick={onClick} />
          <Statistics data={statistics} />
        </div>
      </div>
    );
  }
}

export default User;
