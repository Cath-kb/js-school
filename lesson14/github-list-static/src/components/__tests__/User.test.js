import React from 'react';
import ReactDOM from 'react-dom';
import renderer from "react-test-renderer";
import User from './../User';
import users from './../__fixtures__/users';

const user = users[1];

describe('User', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<User user={user} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should be rendered without errors', () => {
    const html = renderer.create(<User user={user} />).toJSON();
    expect(html).toMatchSnapshot();
  });
});
