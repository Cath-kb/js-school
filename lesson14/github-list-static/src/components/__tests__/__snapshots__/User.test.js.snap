// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should be rendered without errors 1`] = `
<div
  className="row my-4"
>
  <div
    className="col-sm-2"
    data-id="user"
  >
    <img
      alt="User Avatar"
      className="w-100 img-fluid rounded mb-4 mb-sm-0"
      src="https://avatars0.githubusercontent.com/u/19?v=4"
    />
  </div>
  <div
    className="col-sm-10"
  >
    <h2
      className="d-flex align-items-center"
    >
      brynary
      <span
        className="ml-4 badge badge-pill badge-dark"
      >
        admin
      </span>
    </h2>
    <h3>
      <a
        href="https://api.github.com/users/brynary"
        target="_blank"
        title="Go to user profile"
      >
        https://api.github.com/users/brynary
      </a>
    </h3>
  </div>
</div>
`;
